src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/task/taskHelpers.ts

```diff
@@ -56,7 +63,18 @@ export async function getTaskForTool(id: string | undefined, taskDefinition: { t
 	if (!task) {
 		return;
 	}
-	const configuringTasks: IStringDictionary<ConfiguringTask> | undefined = (await taskService.getWorkspaceTasks())?.get(URI.file(workspaceFolder).toString())?.configurations?.byIdentifier;
+
+	let tasksForWorkspace;
+	for (const [folder, tasks] of workspaceFolderToTaskMap) {
+		if (URI.parse(folder).path === workspaceFolder) {
```

Comment:
Having `workspaceFolder` as a string here isn't ideal, comparing 2 URIs should be safer